tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -1056,6 +1057,21 @@ private void writePBTextFile(DAG dag) {
     }
   }
 
+  private void writePBJsonFile(DAGPlan dagPB, DAGImpl newDag) {
+    String logFile = logDirs[new Random().nextInt(logDirs.length)] + File.separatorChar + newDag.getID() + "-"
+        + TezConstants.TEZ_PB_PLAN_JSON_NAME;
+
+    LOG.info("Writing DAG JSON plan to: " + logFile);
+    File outFile = new File(logFile);
+    try {
+      PrintWriter printWriter = new PrintWriter(outFile, "UTF-8");
+      printWriter.println(DAGUtils.generateSimpleJSONPlan(dagPB));
+      printWriter.close();
```

Comment:
I think this close should be in ```finally``` block or we should use try-with-resource for `printWriter`, else if `` printWriter.println(DAGUtils.generateSimpleJSONPlan(dagPB));`` this throws exception, we won't be closing the `printWritter`